apm-agent-core/src/main/java/co/elastic/apm/agent/bci/classloading/DiscriminatingMultiParentClassLoader.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.bci.classloading;
+
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+class DiscriminatingMultiParentClassLoader extends ClassLoader {
```

Comment:
What's the justification of removing `IndyPluginClassLoaderParent` and copying most logic to this one?

Reply:
It's now used in two different contexts see `IndyPluginClassLoader.getParent`.

The way I look at it, I did not remove `IndyPluginClassLoaderParent`. Instead, I made the implementation more generic. I think the name `DiscriminatingMultiParentClassLoader` is preferable because it reflects what the class loader does more closely. The class loader doesn't need to know that it's used as a parent in the `IndyPluginClassLoader`.